### PR TITLE
feat(replays): Change tab navigation to use URL

### LINKED ISSUE #34214: Set the current tab based on the url at pageload, instead of defaulting to 'performance' every time

### PR DESCRIPTION
**Pull request by billyvg**
This changes the tab navigation to use the current state of URL instead of React state. Fixes selecting the correct tab on a page load.

Fixes #34214

### CODE REVIEW

File: static/app/views/replays/detail/focusArea.tsx [hunk 1]
```diff
@@ -18,13 +19,22 @@ type Props = {
   memorySpans: MemorySpanType[] | undefined;
 };
 
+const DEFAULT_TAB = 'performance';
+const TABS = new Set(['performance', 'errors', 'tags', 'memory']);
```
Comment (ryan953) [inline static/app/views/replays/detail/focusArea.tsx:23]: turning `TabBarId` into a ts enum would make it so that we don't need to type the `new Set` here.

tl/dr:
```
// transforms into an object: `{CONSOLE: 'console', ... }`
enum TABS {
  CONSOLE = 'console',
  PERFORMANCE = 'performance',
  ERRORS = 'errors',
  TAGS = 'tags',
  MEMORY = 'memory',
};

function isTabBarId(tab: string): tab is TABS {
  return tab.toUpperCase() in TABS;
}
```

[ts playground](https://www.typescriptlang.org/play?#code/PTAEBcE8AcFNQCoEMBGAhJAnAkgE1ALygDkAxgPYB2AzuQDazGgA+JcmAZuZgLZKWlGLErEyZu1Jq2LgkAc0nDiPWD26RiAbgBQIUBRrhEAQTQBlQqEqwA7qDOxwACgDaZKrQbEANG1FdefkEfETEJEJl5SV9lVXViAF0ASh1tWEoAVx4Tc1AAb21QUABhAHkAOTNSgBkAUUt3GnpGb0LQAAVagCUAMVKugFljcuL6omJ2AL4BFrburv6LcdFxTGi2hGMAcSWSWQUfNoHagf6ATQaVNUwNVoBfVI4MgXAASypQV+pkdCw8J1kKAAXKBqOBMK9KHIkiDAZ9qDkLAUiphHBlMJQIKgAHTgcgAVWg7GKSGosCcSU+mM25h0d202gMYNAAAtSSzLDZIbhyDZsXRyKQkG8qNi2dQWdjUdA6EhBE5iABiCLEFKMjxGQE9cQ8AAS7MsXx+GBwuCc4pZlIA-KyDSCaWZsZ1ev0hiNao9nqQRZiuOQAah7aYzJTkfoPM1+eQ5Aq4V8QoC1fTtH6AyhteQ9ey1amHdiypUarUUkA

)

File: static/app/views/replays/detail/focusTabs.tsx [hunk 2]
```diff
@@ -9,40 +9,25 @@ import {TabBarId} from '../types';
...
-          {t('Tags')}
-        </a>
+        <a href="#tags">{t('Tags')}</a>
       </li>
       <li className={active === 'memory' ? 'active' : ''}>
-        <a href="#memory" onClick={select('memory')}>
-          {t('Memory')}
-        </a>
+        <a href="#memory">{t('Memory')}</a>
       </li>
```
Comment (ryan953) [inline static/app/views/replays/detail/focusTabs.tsx:31]: With the Set or enum I'm tempted to turn this into a loop over the options. Should be like 5 lines or whatever, could inline it into focusArea.tsx

**Review by ryan953 — APPROVED** [review]
some ideas, nothing blocking.

shipit!

File: static/app/views/replays/detail/focusTabs.tsx (hunk 2 above)
Comment (billyvg) [reply static/app/views/replays/detail/focusTabs.tsx:31]: Yeah let's do that.

**Review by ryan953 — APPROVED** [review]
